selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/NewWorkspacePageTest.java

```diff
@@ -0,0 +1,519 @@
+          "nodejs4",
+          "wildfly-swarm",
+          "ceylon-java-javascript-dart-centos",
+          "debian",
+          "debianlsp",
+          "che-in-che",
+          "vert.x",
+          "go-default",
+          "hadoop-default",
+          "java-centos",
+          "java-debian",
+          "kotlin-default",
+          "node-default",
+          "openshift-default",
+          "php-default",
+          "php-gae",
+          "php5.6-default",
+          "platformio",
+          "python-default",
+          "python-2.7",
+          "python-gae",
+          "rails-default",
+          "selenium",
+          "spring-boot",
+          "tomee-default",
+          "ubuntu",
+          "zend");
+
+  private static final List<String> EXPECTED_ALL_STACKS =
+      Arrays.asList(
+          "blank-default",
+          "java-default",
+          "java-mysql",
+          "dotnet-default",
+          "android-default",
+          "cpp-default",
+          "centos",
+          "centos-go",
+          "nodejs4",
+          "wildfly-swarm",
+          "ceylon-java-javascript-dart-centos",
+          "debian",
+          "debianlsp",
+          "che-in-che",
+          "vert.x",
+          "go-default",
+          "hadoop-default",
+          "java-centos",
+          "java-debian",
+          "java-theia-docker",
+          "java-theia-openshift",
+          "java-centos-mysql",
+          "kotlin-default",
+          "node-default",
+          "openshift-default",
+          "openshift-sql",
+          "php-default",
+          "php-gae",
+          "php5.6-default",
+          "platformio",
+          "python-default",
+          "python-2.7",
+          "python-gae",
+          "rails-default",
+          "selenium",
+          "spring-boot",
+          "tomee-default",
+          "ubuntu",
+          "zend");
+
+  private static final List<String> EXPECTED_MULTI_MACHINE_STACKS =
+      asList(
+          "java-mysql",
+          "java-theia-docker",
+          "java-theia-openshift",
+          "java-centos-mysql",
+          "openshift-sql");
+
+  private static final List<String> EXPECTED_QUICK_START_STACKS_REVERSE_ORDER =
+      asList(
+          "blank-default",
+          "java-default",
+          "dotnet-default",
+          "rails-default",
+          "python-default",
+          "platformio",
+          "php-default",
+          "openshift-sql",
+          "openshift-default",
+          "node-default",
+          "java-theia-openshift",
+          "go-default",
+          "che-in-che",
+          "cpp-default",
+          "android-default");
+
+  private static final List<String> EXPECTED_JAVA_STACKS =
+      asList("java-default", "android-default", "che-in-che", "java-theia-openshift");
+
+  private static final List<String> EXPECTED_FILTERS_SUGGESTIONS =
+      asList("JAVA", "JDK", "JAVA 1.8, THEIA");
+
+  private static final List<String> VALID_NAMES =
+      asList("Wk-sp", "Wk-sp1", "9wk-sp", "5wk-sp0", "Wk19sp", "Wksp-01");
```

Comment:
looks like names duplicates each other and it's not quite clear while we chose them to validate the names

Reply:
Different combinations of letters and digits, this approach described in the test plan https://docs.google.com/document/d/1aXUAWFOOylrk-_nO5Iqm77M0Viu8ykCjaQMtJuEBvUA/edit